### DISCUSSION TITLE
Use logfile path / name in the action?

### DISCUSSION THREAD

**Discussion by 65** [fail2ban/fail2ban#3350] [Q&A] [answered]
We are running fail2ban on webservers that are monitoring several log files of individual vhosts in one action. 

Files are named ```/var/log/ols/domain.com.access.log```

The jail.local looks like this: 

```
[wordpress-wp-login]
enabled = true
port = http,https
filter = wordpress-wp-login
logpath = /var/log/ols/*access.log
maxretry = 5
bantime = 1200
ignoreip =
```

I need to know which log file the jail is being triggered for in the action. 

We know that if we changed the log file fields we could pass through for example %v from LiteSpeed (or $http_host for nginx) - however our hosting provider is not able to persist these changes for us for the foreseeable future. 

Is it possible to pass through the path of the monitored logfile into the action as a variable? 

e.g. 

actionban = < LogFilePath >

**Comment by sebres** [comment, marked as answer]
Unfortunately not - the information about logfile where it is occurred isn't stored in the ticket at the moment.
One can surely patch fail2ban (basically enough to patch `fail2ban/server/filter.py`) to add this, here:
<details><summary>patch ...</summary>

```diff
diff --git a/fail2ban/server/filter.py b/fail2ban/server/filter.py
--- a/fail2ban/server/filter.py
+++ b/fail2ban/server/filter.py
@@ -698,7 +698,7 @@ class Filter(JailThread):
                self.processedLine = lambda: "".join(tupleLine[::2])
                return self.findFailure(tupleLine, date, noDate=noDate)

-       def processLineAndAdd(self, line, date=None):
+       def processLineAndAdd(self, line, date=None, filename=None):
                """Processes the line for failures and populates failManager
                """
                try:
@@ -708,6 +708,7 @@ class Filter(JailThread):
                                # ensure the time is not in the future, e. g. by some estimated (assumed) time:
                                if self.checkFindTime and unixTime > MyTime.time():
                                        unixTime = MyTime.time()
+                               if filename: fail['log'] = filename
                                tick = FailTicket(ip, unixTime, data=fail)
                                if self._inIgnoreIPList(ip, tick):
                                        continue
@@ -1146,7 +1147,7 @@ class FileFilter(Filter):
                                                break
                                        # acquire in operation from log and process:
                                        self.inOperation = inOperation if inOperation is not None else log.inOperation
-                                       self.processLineAndAdd(line)
+                                       self.processLineAndAdd(line, filename=filename)
                finally:
                        log.close()
                if self.jail.database is not None:
```
</details>

and then one would be able to use tag `<F-LOG>` in action, which would contain **last** log filename.
Since the jail monitoring multiple logs the IP may be found in several logs (`maxretry` times during the `findtime`), the "last" means hereby that only the last would win. With other words you don't really know where the failures coming from (for instance by `maxretry = 5`, they can be 4 attempts from `a.log` and last 1 attempt from `b.log`, and you'd find `b.log` in the action.

Another variant would be to monitor single file each jail, so you can use `<name>` in the action (or supply logpath to the action as parameter).
In this case it is surely unique.

**Reply by 65** [reply]
Appreciate the detailed answer, thanks. I won't be patching core files, so we'll look at other non-invasive methods to achieve this.
